src/os/bluestore/bluefs_types.h

```diff
@@ -87,8 +87,13 @@ struct bluefs_fnode_t {
   }
 
   void append_extent(const bluefs_extent_t& ext) {
-    extents_index.emplace_back(allocated);
-    extents.push_back(ext);
+    if (!extents.empty() &&
+	extents.back().end() == ext.offset) {
+      extents.back().length += ext.length;
```

Comment:
being a bit paranoic - shouldn't we check for 32-bit length overflow here?

Reply:
ifed01: I agree, at least from a theoretical standpoint we can overflow here.